src/include/sof/ipc.h

```diff
@@ -202,14 +202,25 @@ int ipc_dma_trace_send_position(void);
 /* get posn offset by pipeline. */
 int ipc_get_posn_offset(struct ipc *ipc, struct pipeline *pipe);
 
+enum sof_pm_state {
+	SOF_PM_D0 = 0,	/* Transition to D0, also initial state */
+	SOF_PM_D3,	/* Transition to D3 is requested */
+	SOF_PM_D0ix,	/* Transitions across D0ix substates , incl. D0 are allowed */
+};
+
+enum sof_pm_streaming {
```

Comment:
Does this come from host -> DSP ? Wondering if flags would be better as we may permit/deny other operations in the future (not just audio streams) ?

Reply:
I've updated the structure just to include only generic fields to disable certain power optimizations. FW will be checking if only pipelines related to keyword detection feature are running (no active streaming to host effectively) and then will opt in for Low Power Sequence (LPS) flow.